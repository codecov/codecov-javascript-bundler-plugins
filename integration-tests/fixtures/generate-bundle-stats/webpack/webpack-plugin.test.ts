/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import { $ } from "bun";
import { describe, it, expect, afterEach, beforeEach } from "bun:test";

const webpackPath = (version: number) =>
  `node_modules/webpackV${version}/bin/webpack.js`;
const webpackConfig = (version: number, format: string) =>
  `fixtures/generate-bundle-stats/webpack/webpack-v${version}-${format}.config.cjs`;
const webpackApp = "test-apps/webpack";
const genConfigScript = "scripts/gen-config.sh";

const VERSIONS = [5];
const FORMATS = ["array-push", "commonjs", "module"];

describe("Generating webpack stats", () => {
  describe.each(VERSIONS)(`%d`, (version) => {
    describe.each(FORMATS)(`%s`, (format) => {
      beforeEach(async () => {
        await $`sh ${genConfigScript} webpack commonjs ${format} v5 v${version} cjs`;
      });

      afterEach(async () => {
        await $`rm -rf ${webpackConfig(version, format)}`;
        await $`rm -rf ${webpackApp}/distV${version}`;
      });

      it("matches the snapshot", async () => {
        const id = `webpack-v${version}-${format}-${Date.now()}`;
        const webpack = webpackPath(version);
        const configFile = webpackConfig(version, format);
        const API_URL = `http://localhost:8000/test-url/${id}/200/false`;

        // build the app
        await $`API_URL=${API_URL} node ${webpack} --config ${configFile}`;

        // fetch stats from the server
        const res = await fetch(`http://localhost:8000/get-stats/${id}`);
        const data = (await res.json()) as { stats: string };
        const stats = JSON.parse(data.stats) as unknown;

        expect(stats).toMatchSnapshot({
          builtAt: expect.any(Number),
          duration: expect.any(Number),
          outputPath: expect.stringContaining(`/distV${version}`),
        });
      });
    });
  });
});
