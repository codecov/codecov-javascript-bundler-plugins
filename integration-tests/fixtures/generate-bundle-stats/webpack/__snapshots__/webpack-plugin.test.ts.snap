// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Generating webpack stats 5 {"format":"array-push","expected":"array-push"} matches the snapshot 1`] = `
{
  "assets": [
    {
      "gzipSize": 25254,
      "name": "main-6c1d26e76f6ba1fc75c8.js",
      "normalized": "main-*.js",
      "size": 70961,
    },
  ],
  "builtAt": Any<Number>,
  "bundleName": StringContaining "test-webpack-v5-array-push",
  "bundler": {
    "name": "webpack",
    "version": "5.90.0",
  },
  "chunks": [
    {
      "entry": true,
      "files": [
        "main-6c1d26e76f6ba1fc75c8.js",
      ],
      "id": "179",
      "initial": true,
      "names": [
        "main",
      ],
      "uniqueId": "0-179",
    },
  ],
  "duration": Any<Number>,
  "modules": [
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "./test-apps/webpack/src/main.js + 1 modules",
      "size": 395,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "../node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js",
      "size": 544098,
    },
    {
      "chunkUniqueIds": [],
      "name": "./test-apps/webpack/src/getRandomNumber.js",
      "size": 106,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/global",
      "size": 221,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/node module decorator",
      "size": 123,
    },
  ],
  "outputPath": StringContaining "/distV5",
  "plugin": {
    "name": StringMatching "@codecov/webpack-plugin",
    "version": "0.0.1-beta.12",
  },
  "version": "2",
}
`;

exports[`Generating webpack stats 5 {"format":"commonjs","expected":"cjs"} matches the snapshot 1`] = `
{
  "assets": [
    {
      "gzipSize": 25254,
      "name": "main-6c1d26e76f6ba1fc75c8.js",
      "normalized": "main-*.js",
      "size": 70961,
    },
  ],
  "builtAt": Any<Number>,
  "bundleName": StringContaining "test-webpack-v5-cjs",
  "bundler": {
    "name": "webpack",
    "version": "5.90.0",
  },
  "chunks": [
    {
      "entry": true,
      "files": [
        "main-6c1d26e76f6ba1fc75c8.js",
      ],
      "id": "179",
      "initial": true,
      "names": [
        "main",
      ],
      "uniqueId": "0-179",
    },
  ],
  "duration": Any<Number>,
  "modules": [
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "./test-apps/webpack/src/main.js + 1 modules",
      "size": 395,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "../node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js",
      "size": 544098,
    },
    {
      "chunkUniqueIds": [],
      "name": "./test-apps/webpack/src/getRandomNumber.js",
      "size": 106,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/global",
      "size": 221,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/node module decorator",
      "size": 123,
    },
  ],
  "outputPath": StringContaining "/distV5",
  "plugin": {
    "name": StringMatching "@codecov/webpack-plugin",
    "version": "0.0.1-beta.12",
  },
  "version": "2",
}
`;

exports[`Generating webpack stats 5 source maps are enabled does not include any source maps 1`] = `
{
  "assets": [
    {
      "gzipSize": 25254,
      "name": "main-6c1d26e76f6ba1fc75c8.js",
      "normalized": "main-*.js",
      "size": 70961,
    },
  ],
  "builtAt": Any<Number>,
  "bundleName": StringNotContaining ".map",
  "bundler": {
    "name": "webpack",
    "version": "5.90.0",
  },
  "chunks": [
    {
      "entry": true,
      "files": [
        "main-6c1d26e76f6ba1fc75c8.js",
      ],
      "id": "179",
      "initial": true,
      "names": [
        "main",
      ],
      "uniqueId": "0-179",
    },
  ],
  "duration": Any<Number>,
  "modules": [
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "./test-apps/webpack/src/main.js + 1 modules",
      "size": 395,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "../node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js",
      "size": 544098,
    },
    {
      "chunkUniqueIds": [],
      "name": "./test-apps/webpack/src/getRandomNumber.js",
      "size": 106,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/global",
      "size": 221,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/node module decorator",
      "size": 123,
    },
  ],
  "outputPath": StringContaining "/distV5",
  "plugin": {
    "name": StringMatching "@codecov/webpack-plugin",
    "version": "0.0.1-beta.12",
  },
  "version": "2",
}
`;

exports[`Generating webpack stats 5 {"format":"module","expected":"esm"} matches the snapshot 1`] = `
{
  "assets": [
    {
      "gzipSize": 25254,
      "name": "main-6c1d26e76f6ba1fc75c8.js",
      "normalized": "main-*.js",
      "size": 70961,
    },
  ],
  "builtAt": Any<Number>,
  "bundleName": StringContaining "test-webpack-v5-esm",
  "bundler": {
    "name": "webpack",
    "version": "5.90.0",
  },
  "chunks": [
    {
      "entry": true,
      "files": [
        "main-6c1d26e76f6ba1fc75c8.js",
      ],
      "id": "179",
      "initial": true,
      "names": [
        "main",
      ],
      "uniqueId": "0-179",
    },
  ],
  "duration": Any<Number>,
  "modules": [
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "./test-apps/webpack/src/main.js + 1 modules",
      "size": 395,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "../node_modules/.pnpm/lodash@4.17.21/node_modules/lodash/lodash.js",
      "size": 544098,
    },
    {
      "chunkUniqueIds": [],
      "name": "./test-apps/webpack/src/getRandomNumber.js",
      "size": 106,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/global",
      "size": 221,
    },
    {
      "chunkUniqueIds": [
        "0-179",
      ],
      "name": "webpack/runtime/node module decorator",
      "size": 123,
    },
  ],
  "outputPath": StringContaining "/distV5",
  "plugin": {
    "name": StringMatching "@codecov/webpack-plugin",
    "version": "0.0.1-beta.12",
  },
  "version": "2",
}
`;
